### ISSUE TITLE
OpenGL not working in Proton

### ISSUE THREAD

**Issue by jansol** [NixOS/nixpkgs#148483]
### Describe the bug
CloudBuilt (AppID 262390) fails to launch. It uses OpenGL but does not have a native Linux build so it runs via Proton. I can't remember any other game off the top of my head that I know to use OpenGL 3.2 and not have a Linux build but I imagine they would also be affected.

Some suspicious lines from the journal right before Proton throws an "abort/retry/ignore" dialog and the game dies:
```
pressure-vessel-wrap[117646]: W: Found more than one possible libdrm data directory from provider
SDL_CreateWindow Error: No matching GL pixel format available
SDL_GetWindowWMInfo Error: Invalid window
Not Windows, this should be handledSDL_GetWindowWMInfo Error: Invalid window
Failed to initialize OpenGL
Initiated OpenGL
Error: OpenGL 3.2 not supported
```

### Additional context
This used to be working just fine, some change in the last 6 months broke it. Also it works with Proton 5.0 so it seems to be specifically changes related to running Proton with the newer Soldier runtime.

### Notify maintainers
@jonringer 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.14.21, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-22.05pre334855.29d1f6e1f62"`
 - channels(jansol): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: steam
# a list of nixos modules affected by the problem
module:
```

**Comment by from-nibly** [comment]
Also having this same issue specifically with the game freeways.

**Comment by tfc** [comment]
i also get a similar error message where i think it's the same problem with the "unmetal" demo via steam. It says "ChoosePixelFormat failed" and googling that error message hints towards a lack of opengl contexts.

**Comment by AtilaSaraiva** [comment]
I'm having the same problem with the game Move or Die. I started using the flatpak version and it worked fine.

**Comment by jonringer** [comment]
Is this still relevant? I have made many fixes since this was opened.

**Comment by AtilaSaraiva** [comment]
Oh nice, I will test it out.

**Comment by jansol** [comment]
This particular problem persists on nixos-unstable as of [1ffba9f2f683](https://github.com/NixOS/nixpkgs/commit/1ffba9f2f683063c2b14c9f4d12c55ad5f4ed887)

**Comment by jonringer** [comment]
which video card manufacturer do you have? for nvidia, can you try enabling modeset?

**Comment by jansol** [comment]
The card is an AMD Radeon RX 6600XT, using the mesa driver.

**Comment by jansol** [comment]
Still broken, also affects the Cyberpunk 2077 launcher

**Comment by SuperSamus** [comment]
Other than confirming the problem, there is another issue with both Proton and regular Wine: they are unable to use WineD3D (from Direct3D games to OpenGL).

You can test on Proton with by adding the launch options: `PROTON_USE_WINED3D=1 %command%`.
You can test on Wine by... not installing DXVK (which is necessary because of this issue).

However, Windows games that are already OpenGL do work on Wine (unlike Proton).

Using an AMG iGPU with Mesa.

**Comment by halturin** [comment]
not  sure if its related, but got this error on trying to start Cyberpunk 2077
![image](https://user-images.githubusercontent.com/118860/178251525-5c9cd315-2459-49a5-b393-9e88476fc22d.png)

**Comment by plabadens** [comment]
The CDPR launcher requires OpenGL, even though the game itself doesn't. You can work around the launcher crashing by adding `--launcher-skip` to the Steam command line options, and that should start the game directly.

**Comment by jansol** [comment]
That does indeed work around the problem for that game, but the launcher is an easy way to reproduce the issue with a decently popular title so it is good to have listed here.

**Comment by halturin** [comment]
I think I know why it can't reach these libs...
```
❯❯❯❯ steam                                                                                                          
steam.sh[180052]: Running Steam on nixos 22.11 64-bit                                                               
steam.sh[180052]: STEAM_RUNTIME is enabled automatically                                                            
setup.sh[180135]: Steam runtime environment up-to-date!                                                             
run.sh[180147]: Warning: An unexpected error occurred while executing "/sbin/ldconfig -XNv", the exit status was 127
steam.sh[180052]: Steam client's requirements are satisfied                                                         
```
`run.sh` must be patched.

**Comment by jansol** [comment]
I suspect this is about the ldconfig database again. There was a lot of fiddling with that in the past too.

**Comment by ashkitten** [comment]
so, i messed around a bit with this issue today. it turns out, adding `/run/opengl-driver{,32}/lib` to `ld.so.conf` seems to work... for 64-bit games (i tested with cs:go)

running `ldconfig` from `steam-run` spews out a bunch of stuff about libraries being truncated:
<details>
<pre>
$ ldconfig
/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: Can't link /lib/p11-kit-trust.so to libnssckbi.so
/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /lib32/libGLX.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /lib32/libGLX.so.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /lib32/libGLX.so.0.0.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /lib32/libSDL2-2.0.so.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /lib32/libSDL2-2.0.so.0.22.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /lib32/libSDL2.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: Can't link /lib32/p11-kit-trust.so to libnssckbi.so
/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libEGL_mesa.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libEGL_mesa.so.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libEGL_mesa.so.0.0.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libGLX_mesa.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libGLX_mesa.so.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libGLX_mesa.so.0.0.0 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libvulkan_intel.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libvulkan_lvp.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libvulkan_radeon.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libxatracker.so is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libxatracker.so.2 is truncated

/nix/store/rb2yfmzdciahc6czkh7s4gaj8m56zkbr-glibc-2.34-210-bin/bin/ldconfig: file /run/opengl-driver-32/lib/libxatracker.so.2.5.0 is truncated
</pre>
</details>

i'm not sure why it's saying this; it seems similar to [this issue](https://github.com/NixOS/patchelf/issues/44), but that issue was resolved in glibc 2.31 (we're on 2.34), and this also only seems to affect 32-bit libraries. i tried to run `ldconfig` under a debugger, but due to optimization i wasn't able to figure out anything useful even with symbols from `glibc.debug`.

**Comment by ashkitten** [comment]
https://github.com/NixOS/nixpkgs/pull/184842 should fix this issue. please test!
